source/common/common/generate_version_linkstamp.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# This script generates a header file that is used by version_lib whenever linkstamp is not allowed.
+# linkstamp is used to link in version_linkstamp.cc into the version_lib.
+# However, linkstamp is not available to non-binary bazel targets.
+# This means that if the topmost target being used to compile version_lib is a envoy_cc_library or related, linkstamp will not be in effect.
+# In turn this means that version_linkstamp.cc is not linked, and the build_scm_revision and build_scm_status are unknown symbols to the linker.
+build_scm_revision=$(grep BUILD_SCM_REVISION bazel-out/volatile-status.txt | sed 's/^BUILD_SCM_REVISION //' | tr -d '\\n')
```

Comment:
How does this work when Envoy is included transitively in a project? I'm thinking something like `envoy-filter-example`, or whatever is being done for Envoy Mobile.

Reply:
As long as the outer project is a git repo or it has a SOURCE_VERSION file it works the same as the current flow via linkstamp, i.e it uses the outer project's get_workspace_status command. In Envoy mobile this is set to Envoy's script -- envoy/bazel/get_workspace_status.